lib/nodejs/test/header.test.js

```diff
@@ -39,6 +39,19 @@ const cases = {
     assert.equals(headers.Trace, "abcde");
     assert.end();
   },
+  "Should read different headers from different payload": function(assert) {
+    const transport = new TFramedTransport();
+    const buf = Buffer.from(headerPayload);
+    buf[24] = 115; // Change Parent to Parens
+    buf[32] = 122; // Change shoobar to shoobaz
+    transport.inBuf = buf;
+
+    const headers = transport.readHeaders();
+    assert.equals(headers.Parent, undefined);
```

Comment:
Just to be 100% clear, before this fix, `headers` would have contained `{ Parent: "shoobar", Parens: "shoobaz:", Trace: "abcde" }` because it was reusing the same object as the first test. This is not test-only though and happens in real life.

Similar story for the write test.